Fitur Edit Transaksi dari Daftar

xyzabdkim opened the pull request:
tombol edit di halaman Daftar, setelah klik Edit, akan diarahkan ke halaman Create, data akan diload seperti:
- tanggal
- code pelanggan
- nama pelanggan
- ongkir
- diskon
setelah beberapa saat, code pelanggan akan dicapture dan akan set alamat yang dituju

list item dan ringkasan bisa ditampilkan, fitur hapus pada list Item juga sudah bisa

Edit Transaksi dari Daftar Transaksi

Edit list item sudah bisa jalan, seperti waktu buat transaksi baru
Sudah bisa submit item setelah configure beberapa hal
di mysql:
- add column update_time di transactions
- deposit & discount nullable, default 0

Masalahnya, setelah submit edit, price per item jadi 0, padahal hasil
console log setelah submit edit price per item muncul sesuai ekspektasi

demo edit: http://localhost:8000/penjualan?customer=DEWANGGA&date=20210223&action=create

Update Menu Daftar
Menampilkan daftar sales yang telah dibuat sebelumnya, untuk sementara diset as belum dibayar. Button yang sudah jalan yaitu button edit, setelah klik edit, akan diarahkan ke halaman Create Sales

Fitur Tambah User di Page Admin
Sudah bisa tambah user baru, form yang wajib diisi adalah username, nama lengkap, tanggal lahir, dan agama
Jika username sudah terdaftar, maka user tidak akan ditambah

Revisi comment 28-Feb-21:
- username input
- html data penjualan

Update Admin Tambah User
DItambahkan validasi username dengan ajax (check user valid)

Add validasi form di page Penjualan dan Admin
Form akan berwarna merah jika dirasa tidak memenuhi kriteria minimum ketika akan update/submit

xyzabdkim committed:
Merge branch 'master' of https://github.com/dewzzjr/angkutgan into pasmungdab

xyzabdkim committed:
update fitur Edit Transaksi, state transaksi di
tombol edit di halaman Daftar, setelah klik Edit, akan diarahkan ke
halaman Create, data akan diload seperti:
- tanggal
- code pelanggan
- nama pelanggan
- ongkir
- diskon
setelah beberapa saat, code pelanggan akan dicapture dan akan
set alamat yang dituju

list item dan ringkasan bisa ditampilkan, fitur hapus pada list Item
juga sudah bisa

yang belum bisa:
- edit list item
- submit hasil edit

xyzabdkim committed:
Edit Transaksi dari Daftar Transaksi

Edit list item sudah bisa jalan, seperti waktu buat transaksi baru
Sudah bisa submit item setelah configure beberapa hal
di mysql:
- add column update_time di transactions
- deposit & discount nullable, default 0

Masalahnya, setelah submit edit, price per item jadi 0, padahal hasil
console log setelah submit edit price per item muncul sesuai ekspektasi

xyzabdkim committed:
Merge branch 'master' of https://github.com/dewzzjr/angkutgan into pasmungdab

xyzabdkim committed:
Update Menu Daftar
Menampilkan daftar sales yang telah dibuat sebelumnya, untuk sementara
diset as belum dibayar. Button yang sudah jalan yaitu button edit,
setelah klik edit, akan diarahkan ke halaman Create Sales

xyzabdkim committed:
Fitur Tambah User di Page Admin

Sudah bisa tambah user baru, form yang wajib diisi adalah username, nama
lengkap, tanggal lahir, dan agama
Jika username sudah terdaftar, maka user tidak akan ditambah

dewzzjr commented on frontend/assets/js/admin.js:
```diff
@@ -1,6 +1,142 @@
+const User = {
+  Form: {
+    username: '',
```
validate usernamenya belum ada nih

xyzabdkim replied:
validate seperti apa dew?

awal e pengen cek apakah username udah ada di database apa belum dengan memanfaatkan get users tapi kayane belum ada, sementara diatasi ning handler nek gagal submit artine username udah ada di database

terus secara default iki di bagian username udah ga bisa ngetik spasi, cuma bisa alfanumerik

dewzzjr replied:
kemaren kan tak kasi contoh ini

```
curl --location --request GET 'http://localhost:8000/ajax?action=username&new=test&old=test'
```

xyzabdkim replied:
coba cek latest push dew

dewzzjr commented on frontend/penjualan/daftar.html:
```diff
@@ -21,15 +24,17 @@
+            
+            
+            <!-- <tr>
                 <td>19/12/2020</td>
                 <td>PT. Sejahtera Indah</td>
                 <td>Belum Dibayar</td>
                 <td>
                     <button type="button" class="btn btn-info">Cetak Kwitansi Pembayaran
                         Pelanggan</button>
                     <button type="button" class="btn btn-success">Dibayar</button>
-                    <button type="button" class="btn btn-warning">Ubah</button>
+                    <button type="button" class="btn btn-warning editSales" data-id="DEWANGGA">Ubah</button>
                     <button type="button" class="btn btn-danger">Hapus</button>
                 </td>
             </tr>
```
daftar uda ada endpointnya di commit paling akhir dicoba dulu

xyzabdkim replied:
oiya wes dicoba sebetul e, cuma lali nek sempet hardcode bagian iki

dewzzjr commented on frontend/admin/tambah.html:
```diff
@@ -2,47 +2,66 @@
 		<div class="form-group">
 			<label for="usernameAdd">Username</label>
-			<input name="code" type="text" class="form-control" id="codeAdd" style="text-transform:uppercase" />
+			<input name="code" required type="text" class="form-control" id="codeAdd" style="text-transform:uppercase" />
```
ini dibikin lowercase, nanti pas kirim datanya di sanitize biar ga ada space besar kecil, username yang boleh cuma huruf kecil sama angka aja

xyzabdkim replied:
default ga bisa input space di form nya

xyzabdkim committed:
Revisi comment 28-Feb-21:
- username input
- html data penjualan

xyzabdkim committed:
Merge branch 'master' of https://github.com/dewzzjr/angkutgan into pasmungdab

xyzabdkim committed:
Update Admin Tambah User
DItambahkan validasi username dengan ajax (check user valid)

xyzabdkim committed:
Merge branch 'master' of https://github.com/dewzzjr/angkutgan into pasmungdab

xyzabdkim committed:
Merge branch 'master' of https://github.com/dewzzjr/angkutgan into pasmungdab

xyzabdkim committed:
Add validasi form di page Penjualan dan Admin
Form akan berwarna merah jika dirasa tidak memenuhi kriteria minimum
ketika akan update/submit

dewzzjr commented on frontend/penjualan/daftar.html:
```diff
@@ -21,44 +24,13 @@
+            <!-- JS Event -->
         </tbody>
     </table>
+    <div id='loader' class='text-center'>
+        <img src='https://i.imgur.com/llF5iyg.gif' width='32px' height='32px'>
+        <p>Memuat daftar penjualan...</p>
+    </div>
```
biasakan pake resource yang disediakan, untuk loader uda ada contoh di bootstrap diikutin aja caranya
https://getbootstrap.com/docs/4.4/components/spinners/

dewzzjr reviewed (approved):
sementara ku deploy tapi nanti revisi yo
